core/src/main/java/hudson/util/ConsistentHash.java

```diff
@@ -289,15 +289,18 @@ private synchronized void refreshTable() {
      * Compresses a string into an integer with MD5.
      */
     private int md5(String s) {
-        MD5 md5 = new MD5();
-        md5.update(s.getBytes());
-        byte[] digest = new byte[16];
-        md5.digest(digest);
-
-        // 16 bytes -> 4 bytes
-        for (int i=0; i<4; i++)
-            digest[i] ^= digest[i+4]+digest[i+8]+digest[i+12];
-        return (b2i(digest[0])<< 24)|(b2i(digest[1])<<16)|(b2i(digest[2])<< 8)|b2i(digest[3]);
+        try {
+            MessageDigest md5 = MessageDigest.getInstance("MD5");
+            md5.update(s.getBytes());
+            byte[] digest = md5.digest();
+
+            // 16 bytes -> 4 bytes
+            for (int i=0; i<4; i++)
+                digest[i] ^= digest[i+4]+digest[i+8]+digest[i+12];
+            return (b2i(digest[0])<< 24)|(b2i(digest[1])<<16)|(b2i(digest[2])<< 8)|b2i(digest[3]);
+        } catch (GeneralSecurityException e) {
+            throw new Error("Could not generate MD5 hash", e);
```

Comment:
Does it make sense to throw `Error` here? That seems kind of extreme.

Reply:
maybe is a little excessive, before of changes trilead could return garbage or launch an arithmetic operation exception, probably a runtime exception. Now, digest method could return a https://docs.oracle.com/javase/8/docs/api/java/security/DigestException.html, his supper class is  https://docs.oracle.com/javase/8/docs/api/index.html?java/security/GeneralSecurityException.html, none of these are runtime so add a throws to the method breaks compatibility, the only option is to launch a runtimeException

Reply:
I think some form of RuntimeException would be better than Error.

Reply:
done